keps/sig-auth/3766-referential-authorization/README.md

```diff
@@ -413,76 +403,79 @@ rejected".
 
 ### API Spec
 
+#### ClusterReferenceConsumer
+
 ```golang
-// ReferenceGrant identifies kinds of resources in other namespaces that are
-// trusted to reference the specified kinds of resources in the same namespace
-// as the policy.
-//
-// Each ReferenceGrant can be used to represent a unique trust relationship.
-// Additional ReferenceGrants can be used to add to the set of trusted
-// sources of inbound references for the namespace they are defined within.
-//
-// ReferenceGrant is a form of runtime verification allowing users to assert
-// which cross-namespace object references are permitted. Implementations that
-// support ReferenceGrant MUST NOT permit cross-namespace references which have
-// no grant, and MUST respond to the removal of a grant by revoking the access
-// that the grant allowed. 
-//
-// Implementation of ReferenceGrant is eventually consistent, dependent on
-// watch events being received from the Kubernetes API. Although some processing
-// delay is inevitable, any updates that could result in revocation of access MUST
-// be considered high priority and handled as quickly as possible.
-//
-// Implementations of ReferenceGrant MUST treat all of the following scenarios
-// as equivalent:
-//
-// * A reference to a Namespace that doesn't exist
-// * A reference to a Namespace that exists and a Resource that doesn't exist
-// * A reference to Namespace and Resource that exists but a ReferenceGrant
-//   allowing the reference does not exist
-//
-// If any of the above occur, a generic error message such as "RefNotPermitted" 
-// should be communicated, likely via status on the referring resource.
-//
-// Support: Core
-type ReferenceGrant struct {
+// ClusterReferenceConsumer identifies a common form of referencing pattern. This
+// can then be used with ReferenceGrants to selectively allow references.
+type ClusterReferenceConsumer struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	// Spec defines the desired state of ReferenceGrant.
-	Spec ReferenceGrantSpec `json:"spec,omitempty"`
+	// Subject refers to the subject that is a consumer of the referenced
+	// pattern(s).
+	Subject rbacv1.Subject `json:"subject"`
 
-	// Note that `Status` sub-resource has been excluded at the
-	// moment as it was difficult to work out the design.
-	// A `Status` sub-resource may be added in the future.
+	// The names of the ClusterReferencePatterns this consumer implements.
+	PatternNames []string `json:"patternNames"`
 }
+```
+
+#### ClusterReferencePattern
+
+```golang
+// ClusterReferencePattern identifies a common form of referencing pattern. This
+// can then be used with ReferenceGrants to selectively allow references.
+type ClusterReferencePattern struct {
```

Comment:
some bits above indicate that `verb` should be included here.  Oversight here or mistake there?  Verb makes for interesting possibilities.

Reply:
Good catch, verb was a consideration in earlier versions of this KEP but was intentionally omitted in favor of this translating to `[get, list, watch]` for everything to start. It's possible we may make this configurable in the future. (cc @liggitt for another opinion here)